BookA-Pset_GeoObsBoreholeConstruction

larswik opened the pull request:
# Pset: Pset_GeoObsBoreholeConstruction

## Applicability: IfcBorehole

## Description: 

Properties for describing the construction of a borehole

| Property | Datatype | Description |
|----------|----------|-------------|
|DrillingDiameter|IfcPositiveLengthMeasure|Drilled Borehole diameter (mm)|
|LiningMaterial|IfcLabel|The material of the lining for borehole wall stabilizing|
|BoreholeFilter|IfcLabel|Existence and type or properties of filter segment installed in the casing of a borehole|
|BoreholeInstallation|IfcLabel|Equipments installed in the borehole|

larswik committed:
BookA-Pset_GeoObsBoreholeConstruction

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

SergejMuhic commented on IFC4x3/Properties/b/BoreholeFilter_3fE4eMnCT7eveQISamMQlZ/Documentation.md:
```diff
@@ -0,0 +1 @@
+Existence and type or properties of filter segment installed in the casing of a borehole.
```
Is it just me having understanding issues with this description?

larswik replied:
No. Pinging @j-weil

larswik replied:
I changed to "Identification of filter..."

SergejMuhic commented on IFC4x3/Properties/b/BoreholeInstallation_0o06vNWW901uy$1qJJssW7/Documentation.md:
```diff
@@ -0,0 +1 @@
+Equipments installed in the borehole.
```
Equipment is a noun that knows only singular form.
```suggestion
Equipment installed in the borehole.
```

SergejMuhic commented on IFC4x3/Properties/d/DrillingDiameter_32JL8Vw$L9BhN7Dgy0bzPU/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Is this really needed as a property? Quantity or extracted from geometry should be quite okay here.

larswik replied:
What if the representation is just a line? Ping @j-weil.

SergejMuhic replied:
Understood. I would like to keep this in mind (unless documented somewhere else) that we need a valid representation usage for 'Axis' for `IfcBorehole`. Is it documented or shall I derive an issue from your comment?

larswik replied:
We haved added a usage for this.

SergejMuhic commented on IFC4x3/Properties/l/LiningMaterial_0mAA1gW8z7mevSgdceBIp5/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="LiningMaterial_0mAA1gW8z7mevSgdceBIp5" Name="LiningMaterial" UniqueId="3028a06a-808f-47c2-8e5c-aa79a82d2cc5" PrimaryDataType="IfcLabel" />
```
Propose using `IfcPropertyReferenceValue.PropertyReference` : `IfcMaterialDefinition`

larswik replied:
Fixed in commit https://github.com/bSI-InfraRoom/IFC-Specification/pull/581/commits/f23b2d1e012d12e895e2730c791db8e953e79c53

larswik committed:
Update IFC4x3/Properties/b/BoreholeInstallation_0o06vNWW901uy$1qJJssW7/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Changing LiningMaterial to IfcPropertyreferenceValue + serialization changes

larswik committed:
Merge branch 'feature/BookA-Pset_GeoObsBoreholeConstruction' of https://github.com/larswik/IFC-Specification into feature/BookA-Pset_GeoObsBoreholeConstruction

SergejMuhic commented on IFC4x3/Properties/l/LiningMaterial_0mAA1gW8z7mevSgdceBIp5/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Should this be `IfcPropertyReferenceValue`.`PropertyReference`=`IfcMaterial`?

SergejMuhic reviewed (commented):
Just one question.

larswik committed:
Merge remote-tracking branch 'upstream/tunnel' into feature/BookA-Pset_GeoObsBoreholeConstruction

larswik committed:
According to comments.

Didn't get a response regarding the description of BoreholeFilter. Changed "Existence and type or properties..." to Identification..."
